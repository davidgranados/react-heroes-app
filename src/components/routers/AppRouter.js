import React from "react";
import { BrowserRouter as Router, Switch } from "react-router-dom";
import LoginScreen from "../login/LoginScreen";
import DashboardRoutes from "./DashboardRoutes";
import { PrivateRoute } from "../../routers/PrivateRoute";
import { PublicRoute } from "../../routers/PublicRoute";

const AppRouter = () => {
  return (
    <Router>
      <div>
        <Switch>
          <PublicRoute
            exact
            path="/login"
            component={LoginScreen}
          />
          <PrivateRoute
            path="/"
            component={DashboardRoutes}
          />
        </Switch>
      </div>
    </Router>
  );
};

export default AppRouter;
