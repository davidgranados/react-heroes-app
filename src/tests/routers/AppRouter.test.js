import React from "react";
import { mount } from "enzyme";
import { AuthContext } from "../../auth/AuthContext";
import AppRouter from "../../components/routers/AppRouter";

describe("Pruebas en <AppRouter />", () => {
  const contextValue = {
    authDispatch: jest.fn(),
    authState: {
      logged: false,
    },
  };

  test("debe de mostrar login si no está autenticado", () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar el componente marvel si está autenticado", () => {
    const contextValue = {
      authDispatch: jest.fn(),
      authState: {
        logged: true,
        name: "Juan",
      },
    };

    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect(wrapper.find(".navbar").exists()).toBe(true);
  });
});
