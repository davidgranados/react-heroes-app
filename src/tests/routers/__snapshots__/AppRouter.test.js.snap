// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> debe de mostrar login si no est√° autenticado 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Login
    </h1>
    <hr />
    <form
      onSubmit={[Function]}
    >
      <input
        autoComplete="off"
        className="form-control"
        name="name"
        onChange={[Function]}
        placeholder="username"
        type="text"
        value=""
      />
      <input
        className="form-control"
        name="password"
        onChange={[Function]}
        placeholder="password"
        type="password"
        value=""
      />
      <button
        className="btn btn-primary"
        type="submit"
      >
        Login
      </button>
    </form>
  </div>
</div>
`;
